File: Solution/Python/922. Sort Array By Parity II.py
class Solution:
    def sortArrayByParityII(self, A):
        """
        :type A: List[int]
        :rtype: List[int]
        """
        res = [0 for i in range(len(A))]
        evenIdx = 0
        oddIdx = 1
        for i in A:
            if i % 2 == 0:
                res[evenIdx] = i
                evenIdx += 2
            else:
                res[oddIdx] = i
                oddIdx += 2
        return res


arr = [4, 2, 5, 7]
res = Solution().sortArrayByParityII(arr)
print(res)

File: Solution/Javascript/922. Sort Array By Parity II.js
/**
 * @param {number[]} A
 * @return {number[]}
 */
var sortArrayByParityII = function (A) {
    let res = new Array(A.length);
    let evenIdx = 0;
    let oddIdx = 1;
    for (let i = 0; i < A.length; i++) {
        let cur = A[i];
        if (cur % 2 == 0) {
            res[evenIdx] = cur;
            evenIdx += 2;
        } else {
            res[oddIdx] = cur;
            oddIdx += 2;
        }
    }
    return res;
};


let array = [5, 2, 4, 7]
let res = sortArrayByParityII(array);
console.log(res);